### PR TITLE
fix(wallet): Filter out Zero Balance Groups

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
- Filters out `Network` and `Account` groups that have a `zero` balance on the `Portfolio` page
- Changes the default `Group by` option to `none`
- Add's an `empty` state when there are not tokens to display on the `Portfolio` page

Resolves <https://github.com/brave/brave-browser/issues/31147>
Resolves <https://github.com/brave/brave-browser/issues/31680>

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Go to the `Portfolio` page and open the `Filter and display settings`
2. Uncheck all `Networks` or `Accounts` and click `Save changes`
3. You should see the `Empty Token List` state
4. Test all of the `Call to action` buttons and make sure they work (Deposit, Buy crypto, Add custom asset)

https://github.com/brave/brave-core/assets/40611140/6a4a65f4-63f2-4eea-be04-92c1d56597d4

### CODE REVIEW

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx [hunk 1]
```diff
@@ -410,15 +416,43 @@ export const TokenLists = ({
...
-      const bBalance = getNetworkFiatValue(b)
-      return bBalance.minus(aBalance).toNumber()
-    })?.map((network) => {
+    return [...networks].filter(
+      (network) => getNetworkFiatValue(network)
+        .gt(0)).sort((a, b) => {
+          const aBalance = getNetworkFiatValue(a)
+          const bBalance = getNetworkFiatValue(b)
+          return bBalance.minus(aBalance).toNumber()
+        })
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx:433]: nit: do you think this looks any easier to read?
```suggestion
    return [...networks].filter((network) => {
      return getNetworkFiatValue(network).gt(0)
    }).sort((a, b) => {
      const aBalance = getNetworkFiatValue(a)
      const bBalance = getNetworkFiatValue(b)
      return bBalance.minus(aBalance).toNumber()
    })
```

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx [hunk 2]
```diff
@@ -451,20 +485,45 @@ export const TokenLists = ({
...
       return undefined
     }
-    return [...accounts].sort((a, b) => {
-      const aBalance = getAccountFiatValue(a)
-      const bBalance = getAccountFiatValue(b)
-      return bBalance.minus(aBalance).toNumber()
-    })?.map((account) => {
+    return [...accounts].filter(
+      (account) => getAccountFiatValue(account)
+        .gt(0))
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx:500]: nit: does wrapping this in curly-braces make it any easier to read?

**Review by josheleonard — APPROVED** [review]

**Review by darkdh — APPROVED** [review]
l10n strings ++

File: components/brave_wallet_ui/assets/svg-icons/empty-state-icons/index.ts [hunk 3]
```diff
@@ -7,10 +7,14 @@ import NoAccountsIconLight from './no-accounts-icon-light.svg'
 import NoAccountsIconDark from './no-accounts-icon-dark.svg'
 import NoTransactionsIconLight from './no-transactions-icon-light.svg'
 import NoTransactionsIconDark from './no-transactions-icon-dark.svg'
+import NoTokensIconLight from './no-tokens-icon-light.svg'
```
Comment (BigPolo8519) [inline components/brave_wallet_ui/assets/svg-icons/empty-state-icons/index.ts:10]: - I going to advise u 2 return every penny @

**Pull request merged**